### ISSUE TITLE
Too big current in PreCharge

### ISSUE THREAD

**Issue by mrozigor** [V2GClarity/RISE-V2G#20]
Hello.

I'm not sure if this is a bug or just some thoughts on this topic ;)

Based on IEC 61851-23 (CC6.1 or CC6.2) that I can assume is base for chargers, current in PreCharge phase shouldn't be bigger than 2A. In RV2G target current is set once for whole session. If you would like to implement changes, then RISEV2G/EVCC/WaitForCableCheckRes.java:59 should receive 2A PhysicalValue (maybe another method in controller, e.g. getPreChargeCurrent()?).

Or maybe I'm too detailed at ISO15118 level?

Igor

**Comment by MarcMueltin** [comment]
Sorry for the late reply.
I am not aware of the requirements in IEC 61851-23, but I find it strange that a current value shall not be bigger than 2A. That would mean that DC charging would take ages for an EV with a range of let's say 500 km (assuming lets's say 400 V or even 800 V output voltage).

According to ISO 15118, the PreChargeReq message is used for adjusting the EVSE output voltage to the EV battery voltage. The EVSE adapts the DC output voltage, limits the current accordingly and continuously informs the EV about it in the PreChargeRes messages.

So to my understanding, the targetCurrent (as the targetVoltage) is a fixed value communicated by the EV. You can change that value to your liking in the [DummyEVController](https://github.com/V2GClarity/RISE-V2G/blob/master/RISE-V2G-EVCC/src/main/java/com/v2gclarity/risev2g/evcc/evController/DummyEVController.java#L231).
The DummyEVController is set up to adhere to the correct syntax defined by the underlying XSD schema, but does not necessarily provide meaningful values in all cases. You can play around with these values as you wish.

Does that answer your question or solve your problem?

**Comment by mrozigor** [comment]
Thanks for response.

I know that I can change requested current in controller. For now I changed current to 2A in PreCharge state only. The issue is (or not ;)) with current limitations during PreCharge phase (as I wrote earlier - IEC 618510-23). This limitation only apply to PreCharge, so after voltage adjust, EV can demand higher current value.

On EVSE current higher than 2A can be treated as a out of bond value and therefore charging can be canceled by EVSE. This isn't stated anywhere in ISO15118, so I just create this issue to let you know how chargers can behave ;)
